content/testsuite/provideringester.go

```diff
@@ -0,0 +1,87 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package testsuite
+
+import (
+	"context"
+	"io"
+	"io/ioutil"
+	"testing"
+
+	"github.com/containerd/containerd/content"
+	digest "github.com/opencontainers/go-digest"
+)
+
+// ProviderIngester consists of provider and ingester.
+type ProviderIngester interface {
+	content.Provider
+	content.Ingester
+}
+
+// TestSmallBlob tests a blob which is smaller than the read buffer.
```

Comment:
Why is this test different than all the other checks? Does this need to be exported?

Please follow the existing conventions:

1. This does not need a new file. Please put with the other checks.
2. Don't export the identifiers.

Reply:
Exposed tests in`provideringester.go` is expected to be used for a type that implements `ProviderIngester` but does not fully implement `content.Store` interface.

Updated the code comment for clarity.